### ISSUE TITLE
Chart set_HasAxis fails

### ISSUE THREAD

**Issue by kev-is-coding-for-adn** [NetOfficeFw/NetOffice#376]
Hi,

We are using the last version of NetOffice (1.9.1) and it seems we can't find a solution to simply hide the different axis on a chart.
We have tried to use the **chart.set_HasAxis(XlAxisType.xlCategory, XlAxisGroup.xlPrimary, MsoTriState.msoFalse);** but it fails with the message "Failed to proceed PropertySet on PowerPoint.Chart => HasAxis".

What's working is :
  - var axis = chart.Axes(XlAxisType.xlCategory, XlAxisGroup.xlPrimary) as Axis;
  - axis.Delete();
But then, we can't add it back anymore.

Thanks for your help on this.
Best regards
Kevin

**Comment by jozefizso** [comment]
Hi @kev-is-coding-for-adn,
according to VBA documentation, the `HasAxis` method has only two parameters: https://learn.microsoft.com/en-us/office/vba/api/excel.chart.hasaxis

Are you able to write code as VBA macro and get correct results?

**Comment by kev-is-coding-for-adn** [comment]
Hello,

Thanks for this answer.
Just to mention, we are looking to use NetOffice in PowerPoint.

Unfortunately, the HasAxis method doesn't work in C# as if I type : `chart.HasAxis(XlAxisType.xlCategory, XlAxisGroup.xlPrimary) = MsoTriState.msoFalse;` it says the `The assignment target must be an assignable variable, property, or indexer`.
As you can see in my first post, we have tried to use the 'set_HasAxis' method wich takes 3 parameters.

**Comment by jozefizso** [comment]
That definition of `HasAxis` in NetOffice may not be actually correct, especially when you get the exception when calling it.

To test if Excel/Charts behave as you would like it is best to test the code as VBA macro to see actual behavior.

**Comment by kev-is-coding-for-adn** [comment]
Hmm, I don't know what to answer. I know that in VBA, the `HasAxis(xlValue, xlPrimary) = False` works (tested).
But that same definition doesn't work in C# threw the NetOffice library.

Is that what you meant ?

**Comment by jozefizso** [comment]
Yes, so the method is incorrectly defined in NetOffice.

**Comment by jozefizso** [comment]
Thank for reporting this bug. I have found the problem:

```cs
public void set_HasAxis(object index1, object index2, object value)
{
	Factory.ExecutePropertySet(this, "HasAxis", index1, index2, value);
}

public static void ExecutePropertySet(this Core value, ICOMObject caller, string name, object newValue, object argument1, object argument2)
{ ... }
```

The order of the parameters to the `ExecutePropertySet` method is incorrect. The `newValue` is not set to the `true/false` but to the `index1` parameter.

**Comment by jozefizso** [comment]
The issue is I don't know how many of NetOffice APIs were generated incorrectly and are passing parameters in incorrect order.

**Comment by jozefizso** [comment]
This issue affect NetOffice v1.7, v1.8 and 1.9 releases.

**Comment by jozefizso** [comment]
@kev-is-coding-for-adn can you test this build if it fixes the problem for you?

https://github.com/NetOfficeFw/NetOffice/actions/runs/4033912155

**Comment by kev-is-coding-for-adn** [comment]
Hi @jozefizso 

Tested and approved ! 
Thanks for the quick fix.
Do you think it will be possible to have the same method as VBA, using `HasAxis(xlValue, xlPrimary) = False` ?
For the moment, only `chart.set_HasAxis(XlAxisType.xlCategory, XlAxisGroup.xlPrimary, MsoTriState.msoFalse);` seems to work.

Also, when do you think the 1.9.2 release will be out?

Thanks !
Have a nice week.

**Comment by jozefizso** [comment]
Theoretically, there could be an indexer so in C# the syntax would be `Chart.HaxAxis[xlValue, xlPrimary] = true;` but there is also existing `Chart.HasAxis` property on some of the objects and I need to check if those names would not clash.

I hope the 1.9.2 could be released this week.

**Comment by jozefizso** [comment]
Fixed in v1.9.2.
